### ISSUE TITLE
Make quelpa-build--run-process be able to run asynchronously

### ISSUE THREAD

**Issue by kiennq** [quelpa/quelpa#197]
This will help reduce Emacs down time due to waiting for pulling files from remote repo

**Comment by steckerhalter** [comment]
sounds interesting :smiley:

**Comment by steckerhalter** [comment]
@kiennq do you plan to work on this or shall we close it?

**Comment by steckerhalter** [comment]
Ping

**Comment by kiennq** [comment]
I've put another stab at this issue in the new PR

**Comment by wyuenho** [comment]
@kiennq `url-retrieve-synchronously` will also block, you may want to take another stab :)

**Comment by kiennq** [comment]
> @kiennq `url-retrieve-synchronously` will also block, and throws an error when it times out, you may want to take another stab :)

Yeah, I would keep that as future work since installing from emacs wiki should be much less comparing to installing from git and other svc.
The take would be the same though, wait until url retrieve finish while allow editing via `recursive-edit`. That reduces the code changes scope while still make Emacs interactable (to some extends) while the package is downloading

**Comment by wyuenho** [comment]
That's `url-retrieve-synchronously` call is used to download everything, and it will block for 2 minutes per call if the connection cannot be established before timing out.

**Comment by kiennq** [comment]
> That's `url-retrieve-synchronously` call is used to download everything, and it will block for 2 minutes per call if the connection cannot be established before timing out.

In the quelpa source code, I only saw it's used in `quelpa-build--url-copy-file `, which is used for `quelpa-build--grab-wiki-file`, so only in case of installing from emacs wiki should be affected by that.
For other way of installing package, mostly svc is used to clone the source repo, which is done via external process.

**Comment by wyuenho** [comment]
My mistake, I forgot to look at the rest of the fetchers
